app/controllers/volunteers_controller.rb

```diff
@@ -303,7 +303,16 @@ def home
     @total_shifts_needing_cov = Log.needing_coverage(current_volunteer.region_ids, 7).length
 
     # To Do Pickup Reports
-    @to_do_reports = Log.picked_up_by(current_volunteer.id, false)
+    to_do_reports = Log.picked_up_by(current_volunteer.id, false)
+    puts to_do_reports.size
+    @to_do_reports = Array.new()
+    for to_do in to_do_reports
+      if to_do.when <= Date.today
+        @to_do_reports.push(to_do)
+      end
+      puts @to_do_reports.size
+    end
+
```

Comment:
This `def home` method is kinda huge. Abstract some methods out of it, so it calls on other methods (maybe private methods inside of the controller, or on the Volunteer model), rather than doing all of this work inside of one controller method... I wonder if a lot of the logic under `# To Do Pick Reports` could be moved to the Log model, so all this controller has to do is `Log.get_pickup_reports` or something?